### PR TITLE
Added QueueDoesNotExistException and a test for it

### LINKED ISSUE #1157: Improve log message when Amazon.SQS.Model.QueueDoesNotExistException
Hi,

Today in my non-prod env I'm facing Amazon.SQS.Model.QueueDoesNotExistException and I don't know which queue NServiceBus is trying to access 

The call stack:
```
Amazon.SQS.Model.QueueDoesNotExistException: The specified queue does not exist for this wsdl version.
 ---> Amazon.Runtime.Internal.HttpErrorResponseException: Exception of type 'Amazon.Runtime.Internal.HttpErrorResponseException' was thrown.
  at Amazon.Runtime.HttpWebRequestMessage.GetResponseAsync(CancellationToken cancellationToken)
   at Amazon.Runtime.Internal.HttpHandler`1.InvokeAsync[T](IExecutionContext executionContext)\
   at Amazon.Runtime.Internal.Unmarshaller.InvokeAsync[T](IExecutionContext executionContext)
  at Amazon.SQS.Internal.ValidationResponseHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.ErrorHandler.InvokeAsync[T](IExecutionContext executionContext)
  --- End of inner exception stack trace ---
   at Amazon.Runtime.Internal.HttpErrorResponseExceptionHandler.HandleExceptionStream(IRequestContext requestContext, IWebResponseData httpErrorResponse, HttpErrorResponseException exception, Stream responseStream)
   at Amazon.Runtime.Internal.HttpErrorResponseExceptionHandler.HandleExceptionAsync(IExecutionContext executionContext, HttpErrorResponseException exception)
   at Amazon.Runtime.Internal.ExceptionHandler`1.HandleAsync(IExecutionContext executionContext, Exception exception)
   at Amazon.Runtime.Internal.ErrorHandler.ProcessExceptionAsync(IExecutionContext executionContext, Exception exception)
   at Amazon.Runtime.Internal.ErrorHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.EndpointDiscoveryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.EndpointDiscoveryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CredentialsRetriever.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.RetryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.RetryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.ErrorCallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.MetricsHandler.InvokeAsync[T](IExecutionContext executionContext)
   at NServiceBus.Transport.SQS.QueueCache.GetQueueUrl(String queueName) in /_/src/NServiceBus.Transport.SQS/QueueCache.cs:line 54
   at NServiceBus.Transport.SQS.MessageDispatcher.ApplyUnicastOperationMappingIfNecessary(UnicastTransportOperation transportOperation, SqsPreparedMessage sqsPreparedMessage, Int64 delaySeconds, String messageId, Dictionary`2 nativeMessageAttributes) in /_/src/NServiceBus.Transport.SQS/MessageDispatcher.cs:line 374
   at NServiceBus.Transport.SQS.MessageDispatcher.PrepareMessage[TMessage](IOutgoingTransportOperation transportOperation, HashSet`1 messageIdsOfMulticastedEvents, TransportTransaction transportTransaction) in /_/src/NServiceBus.Transport.SQS/MessageDispatcher.cs:line 322
   at NServiceBus.Transport.SQS.MessageDispatcher.DispatchBatched(IEnumerable`1 toBeBatchedTransportOperations, HashSet`1 messageIdsOfMulticastEvents, TransportTransaction transportTransaction) in /_/src/NServiceBus.Transport.SQS/MessageDispatcher.cs:line 104
   at NServiceBus.Transport.SQS.MessageDispatcher.Dispatch(TransportOperations outgoingMessages, TransportTransaction transaction, ContextBag context) in /_/src/NServiceBus.Transport.SQS/MessageDispatcher.cs:line 59
   at NServiceBus.TransportReceiveToPhysicalMessageConnector.Invoke(ITransportReceiveContext context, Func`2 next) in /_/src/NServiceBus.Core/Pipeline/Incoming/TransportReceiveToPhysicalMessageConnector.cs:line 61
   at NServiceBus.MainPipelineExecutor.Invoke(MessageContext messageContext) in /_/src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs:line 50
   at NServiceBus.Transport.SQS.InputQueuePump.ProcessMessageWithInMemoryRetries(Dictionary`2 headers, String nativeMessageId, Byte[] body, Message nativeMessage, CancellationToken token) in /_/src/NServiceBus.Transport.SQS/InputQueuePump.cs:line 273
```

Maybe it's related with https://github.com/Particular/NServiceBus.AmazonSQS/issues/227

NServiceBus: 7.4.6
NServiceBus.AmazonSQS: 5.4.0
.NET 5.0.403 on Container with Debian

### LINKED ISSUE #1678: The dispatcher doesn't log the missing queue name when the destination queue could not be found
The following is an excerpt from a ServiceControl log. An audit instance needs to send messages to a primary instance. The primary instance queue was not available. From the error message, it's impossible to understand what the missing queue is:

```
2022-10-24 14:15:15.9512|21|Error|NServiceBus.Transport.SQS.MessageDispatcher|Exception from Send.
Amazon.SQS.Model.QueueDoesNotExistException: The specified queue does not exist for this wsdl version. ---> Amazon.Runtime.Internal.HttpErrorResponseException: The remote server returned an error: (400) Bad Request. ---> System.Net.WebException: The remote server returned an error: (400) Bad Request.
```

### PR DESCRIPTION
**Pull request by jpalac**
Fixes #1678
Fixes #1157
Added a catch for QueueDoesNotExistException  and a test for it.

### CODE REVIEW

**Review by mauroservienti — APPROVED** [review]

File: src/NServiceBus.Transport.SQS/MessageDispatcher.cs [hunk 1]
```diff
@@ -363,8 +363,16 @@ async Task ApplyUnicastOperationMappingIfNecessary(UnicastTransportOperation tra
...
                 sqsPreparedMessage.Destination = transportOperation.Destination;
-                sqsPreparedMessage.QueueUrl = await queueCache.GetQueueUrl(sqsPreparedMessage.Destination)
-                    .ConfigureAwait(false);
+
+                try
+                {
+                    sqsPreparedMessage.QueueUrl = await queueCache.GetQueueUrl(sqsPreparedMessage.Destination)
+                        .ConfigureAwait(false);
+                }
+                catch (QueueDoesNotExistException ex)
```
Comment (danielmarbach) [inline src/NServiceBus.Transport.SQS/MessageDispatcher.cs:372]: We can look into this as a dedicated PR. I'm just bringing this up for now. Core has also the [`QueueNotFoundException` type](https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Unicast/Queuing/QueueNotFoundException.cs) that we could use instead

https://github.com/search?q=org%3AParticular+QueueNotFoundException&type=code

**Review by danielmarbach — APPROVED** [review]

**Comment by danielmarbach** [comment]
Will be merged with squash once the build is green. @jpalac @mikeminutillo @mauroservienti let me know your thoughts about using the core exception

File: src/NServiceBus.Transport.SQS/MessageDispatcher.cs (hunk 1 above)
Comment (jpalac) [reply src/NServiceBus.Transport.SQS/MessageDispatcher.cs:372]: We used the QueueDoesNotExistException as that is what is already used in v6 of the transport. It was only v5 that wasn't catching this.
So if we change it, we should be changing it in all the other places that v6 is using it for consistency.

Comment (mauroservienti) [reply src/NServiceBus.Transport.SQS/MessageDispatcher.cs:372]: And changing it might be a breaking change

**Pull request merged**